include/cmm/StructDefinitionStatementNode.h

```diff
@@ -13,15 +13,24 @@
 // Our includes
 #include <cmm/Types.h>
 #include <cmm/BlockNode.h>
+#include <cmm/Field.h>
 #include <cmm/StatementNode.h>
 
 // std includes
+#include <optional>
 #include <string>
+#include <unordered_map>
 
 namespace cmm
 {
     class StructDefinitionStatementNode : public StatementNode
     {
+    public:
+
+        using FieldMap = std::unordered_map<std::string, Field>;
```

Comment:
Consider using new the IField interface.

Reply:
After reviewing the code, I decided to keep as-is to avoid needing to do extra heap allocation.  Any 'getter' type functions will instead use IField, but keep this storage as a "normal" Field.